app/controllers/pensions/RetirementAnnuityController.scala

```diff
@@ -0,0 +1,94 @@
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.PensionSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.pensions.PayIntoRetirementAnnuityContract
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+
+class RetirementAnnuityController @Inject()(implicit val cc: MessagesControllerComponents,
+                                            authAction: AuthorisedAction,
+                                            inYearAction: InYearAction,
+                                            payIntoRetirementAnnuityContract: PayIntoRetirementAnnuityContract,
+                                            appConfig: AppConfig,
+                                            pensionSessionService: PensionSessionService,
+                                            errorHandler: ErrorHandler,
+                                            clock: Clock) extends FrontendController(cc) with I18nSupport with SessionHelper {
+
+
+  def yesNoForm(implicit user: User[_]): Form[Boolean] = YesNoForm.yesNoForm(
+    missingInputError = s"pensions.retirementAnnuityContract.error.noEntry.${if (user.isAgent) "agent" else "individual"}"
+  )
+
+  def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit user =>
+    pensionSessionService.getPensionsSessionDataResult(taxYear) {
+      case Some(data) =>
+        data.pensions.paymentsIntoPension.retirementAnnuityContractPaymentsQuestion match {
+          case Some(value) => Future.successful(Ok(payIntoRetirementAnnuityContract(
+            yesNoForm.fill(value), taxYear)))
+          case None => Future.successful(Ok(payIntoRetirementAnnuityContract(yesNoForm, taxYear)))
+        }
+      case None =>
+        Future.successful(Ok(payIntoRetirementAnnuityContract(yesNoForm, taxYear)))
+    }
+  }
+
+  def submit(taxYear: Int): Action[AnyContent] = authAction.async { implicit user =>
+    pensionSessionService.getPensionsSessionDataResult(taxYear) {
+      data =>
+        yesNoForm.bindFromRequest().fold(
+          formWithErrors => Future.successful(BadRequest(payIntoRetirementAnnuityContract(formWithErrors, taxYear))),
+          yesNo => {
+            val pensionsCYAModel: PensionsCYAModel = data.map(_.pensions).getOrElse(PensionsCYAModel(PaymentsIntoPensionViewModel()))
+            val viewModel: PaymentsIntoPensionViewModel = pensionsCYAModel.paymentsIntoPension
+            val updatedCyaModel: PensionsCYAModel = {
+              if (yesNo) {
+                pensionsCYAModel.copy(paymentsIntoPension = viewModel.copy(retirementAnnuityContractPaymentsQuestion = Some(true)))
+              } else {
+                pensionsCYAModel.copy(paymentsIntoPension = viewModel.copy(retirementAnnuityContractPaymentsQuestion = Some(false)))
+              }
```

Comment:
You could replace the 'if' condition by replacing the 'Some(true)' or 'Some(false)' with 'Some(yesNo)'